src/zope/app/apidoc/codemodule/text.py

```diff
@@ -13,25 +13,24 @@
 ##############################################################################
 """Function representation for code browser
 
-$Id$
 """
 __docformat__ = 'restructuredtext'
-from zope.interface import implements
+from zope.interface import implementer
 from zope.location.interfaces import ILocation
 
 from zope.app.apidoc.codemodule.interfaces import ITextFile
 
+
+@implementer(ILocation, ITextFile)
 class TextFile(object):
     """This class represents a function declared in the module."""
-    implements(ILocation, ITextFile)
 
     def __init__(self, path, name, package):
         self.path = path
         self.__parent__ = package
         self.__name__ = name
 
     def getContent(self):
-        file = open(self.path, 'rU')
-        content = file.read()
-        file.close()
+        with open(self.path, 'rUb') as f:
```

Comment:
Empirically it works, in that it doesn't throw any errors. I don't have any files with odd line-endings to test it on, though. I suspect it's just completely ignored by Python 3 (it's the same as `open(path, 'rb', newline=None)`, where `newline` is presumably ignored for non-text modes but that constructor doesn't throw any exceptions.) I need to try different line endings and see what it does.